### ISSUE TITLE
[Android Card Component] Component asks for security code on recurring payments, despite not being required

### ISSUE THREAD

**Issue by JDSM01** [Adyen/adyen-android#206]
**Describe the bug**

When you make a recurring payments and you don't want to use the CVC/CVV, the component doesn't consider the info valid after you insert the card number and the expiry date.
I've checked the lib and this verification is done on the CardComponent.java, l.260, l261 on the validateSecurityCode method, but since the API (/paymentMethods) never returns the field that the method is checking, they never consider it valid. 
This field is returned on the oneClickPaymentMethods, but since the lib no longer uses it, it's never possible to do a transaction without CVC/CVV
Can you confirm if this is a bug, if it's intended or if I'm doing something wrong?

**To Reproduce**

User makes payment with a card and asks to save it;
On the backend we set everything to enable recurring payments;
On the second purchases the component only considers valid when the user inserts the 3 or 4 CVV/CVC digits

**Expected behavior**
User makes payment with a card and asks to save it;
On the second purchases user is able to do a purchase without the need to insert the CVV/CVC

**Desktop (please complete the following information):**
-Android

**Smartphone (please complete the following information):**

**Additional context**

**Comment by caiofaustino** [comment]
Hey @JDSM01,

So the field is also returned in the `storedPaymentMethods` which is the replacement for the `oneClickPaymentMethods`.
Currently the recurring payments are only handled on the Components for cards that are already stored in our system with a shopper reference, so that way the shopper only has to input the CVC/CVV to finalize the payment. To do that you have to pass the stored payment method instance to the component.

We currently don't support payments without CVC/CVV through the components, only for API only flows such as subscriptions. But I asked internally and will let you know if that changes.

**Comment by JDSM01** [comment]
Thank you for the quick response!

> So the field is also returned in the `storedPaymentMethods` which is the replacement for the `oneClickPaymentMethods`.

Question, when you say that is returned in the `storedPaymentMethods`, I've checked your API and there's nothing related to that on the `storedPaymentMethods`. Does it come in through the `paymentMethods`?

https://docs.adyen.com/api-explorer/#/PaymentSetupAndVerificationService/paymentMethods

> Currently the recurring payments are only handled on the Components for cards that are already stored in our system with a shopper reference, so that way the shopper only has to input the CVC/CVV to finalize the payment. To do that you have to pass the stored payment method instance to the component.

Doing that 

> We currently don't support payments without CVC/CVV through the components, only for API only flows such as subscriptions. But I asked internally and will let you know if that changes.

Ok. Thank you once again.

**Comment by caiofaustino** [comment]
Yes it's part of the `/paymentMethods` response. You need to pass the shopper reference on the call to get the card from that shopper. Here is the documentation:
https://docs.adyen.com/api-explorer/#/PaymentSetupAndVerificationService/v53/post/paymentMethods__resParam_storedPaymentMethods

**Comment by JDSM01** [comment]
Oh, I was talking about the field that is used in the component
```
"key": "cardDetails.cvc",
            "type": "cvc"
```
This field comes in oneClickPaymentMethods, and the component checks for it (type == "cvc") in CardComponent.java, l.260, l261 on the validateSecurityCode method, to do the validation or not of the 3 or 4 digits in the CVV/CVC. And the idea that I had was that when you started using the storedPaymentMethods you forgot that use case, but looks like that functionality was never supported after all.

**Comment by caiofaustino** [comment]
That check is only enforced if the field is present and set as not optional. The cases where it is optional are only some specific card types that might not even have that number, but those are handled separately. So yeah you are right, it's working as intended for the general case.

The CVC is an important part of the payment security, like I said I'll check internally if we have plans to offer a recurring style of flow in the Components and Drop-in in the fututre.
